File: ws/lisintuña/u2/WS23/ComplexOperations/src/main/java/ec/edu/espe/complexoperations/view/ComplexOperations.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package ec.edu.espe.complexoperations.view;

import ec.edu.espe.complexoperations.controller.Operation;
import ec.edu.espe.complexoperations.controller.PersonController;
import java.time.LocalDate;

/**
 *
 * @author Eduardo Segarra, TheJavaBandits, DCCO-ESPE
 */
public class ComplexOperations {

    public static void main(String[] args) {
        System.out.println("Welcome to complex operations! \nEduardo Segarra");
        float addend1;
        float addend2;
        float sum;
        
        //keyboard input
        addend1 = 1;
        addend2 = 2;
        
        sum = Operation.addTwoNumbers(addend1, addend2);
        
        OperationPrinting.printAddition(addend1, addend2, sum);
        
        addend1 = 1.2F;
        addend2 = 2.4F;
        
        sum = Operation.addTwoNumbers(addend1, addend2);
        
        OperationPrinting.printAddition(addend1, addend2, sum);
        
        addend1 = 2.2F;
        addend2 = -1.8F;
        
        sum = Operation.addTwoNumbers(addend1, addend2);
        
        OperationPrinting.printAddition(addend1, addend2, sum);
        
        LocalDate birthDate = LocalDate.parse("2004-01-31");
        int days, years;
        
        days = PersonController.computeAgeInDays(birthDate);
        years = PersonController.computeAgeInYears(birthDate);
        
        System.out.println("Eduardo is " + days + " days old.");
        System.out.println("He was born on " + birthDate);
        
        System.out.println("Eduardo is " + years + "years-old");
    }

}

File: hw/segarra/u2/HW17TestCaseSolution/ComplexOperationsJavaScript/ec.edu.espe.complexoperationsjavascript.view/ComplexOperations.js
const addTwoNumbers = require('../ec.edu.espe.complexoperationsjavascript.model/Operation.js');
const computeAgeInDays = require('../ec.edu.espe.complexoperationsjavascript.model/PersonController.js');
const computeAgeInYears = require('../ec.edu.espe.complexoperationsjavascript.model/PersonController.js');

addend1 = 9.1
addend2 = -3.2
sum = addTwoNumbers(addend1, addend2)

console.log(sum)

birthDate = '31-01-2004'
ageInDays = computeAgeInDays(birthDate)
ageInYears = computeAgeInYears(birthDate)

console.log(ageInDays)
console.log(ageInYears)